doc/src/manual/parallel-computing.md

```diff
@@ -1,12 +1,542 @@
+       end
+
+julia> @elapsed while n > 0 # print out results
+           job_id, exec_time = take!(results)
+           println("$job_id finished in $(round(exec_time,2)) seconds")
+           n = n - 1
+       end
+4 finished in 0.22 seconds
+3 finished in 0.45 seconds
+1 finished in 0.5 seconds
+7 finished in 0.14 seconds
+2 finished in 0.78 seconds
+5 finished in 0.9 seconds
+9 finished in 0.36 seconds
+6 finished in 0.87 seconds
+8 finished in 0.79 seconds
+10 finished in 0.64 seconds
+12 finished in 0.5 seconds
+11 finished in 0.97 seconds
+0.029772311
+```
+
+The current version of Julia multiplexes all tasks onto a single OS thread. Thus, while tasks
+involving I/O operations benefit from parallel execution, compute bound tasks are effectively
+executed sequentially on a single OS thread. Future versions of Julia may support scheduling of
+tasks on multiple threads, in which case compute bound tasks will see benefits of parallel execution
+too.
+
+## @threadcall (Experimental)
+
+All I/O tasks, timers, REPL commands, etc are multiplexed onto a single OS thread via an event
+loop. A patched version of libuv ([http://docs.libuv.org/en/v1.x/](http://docs.libuv.org/en/v1.x/))
+provides this functionality. Yield points provide for co-operatively scheduling multiple tasks
+onto the same OS thread. I/O tasks and timers yield implicitly while waiting for the event to
+occur. Calling [`yield`](@ref) explicitly allows for other tasks to be scheduled.
+
+Thus, a task executing a [`ccall`](@ref) effectively prevents the Julia scheduler from executing any other
+tasks till the call returns. This is true for all calls into external libraries. Exceptions are
+calls into custom C code that call back into Julia (which may then yield) or C code that calls
+`jl_yield()` (C equivalent of [`yield`](@ref)).
+
+Note that while Julia code runs on a single thread (by default), libraries used by Julia may launch
+their own internal threads. For example, the BLAS library may start as many threads as there are
+cores on a machine.
+
+The [`@threadcall`](@ref) macro addresses scenarios where we do not want a [`ccall`](@ref) to block the main Julia
+event loop. It schedules a C function for execution in a separate thread. A threadpool with a
+default size of 4 is used for this. The size of the threadpool is controlled via environment variable
+`UV_THREADPOOL_SIZE`. While waiting for a free thread, and during function execution once a thread
+is available, the requesting task (on the main Julia event loop) yields to other tasks. Note that
+`@threadcall` does not return till the execution is complete. From a user point of view, it is
+therefore a blocking call like other Julia APIs.
+
+It is very important that the called function does not call back into Julia.
+
+`@threadcall` may be removed/changed in future versions of Julia.
+
+# Multi-Threading (Experimental)
+
+In addition to tasks, remote calls, and remote references, Julia forwards natively
```

Comment:
remote calls and remore references are not introduced yet at this point. Sentence needs re-wording.